core/src/components/checkbox/checkbox.tsx

```diff
@@ -253,11 +262,23 @@ export class Checkbox implements ComponentInterface {
 
     renderHiddenInput(true, el, name, checked ? value : '', disabled);
 
+    // Determine appropriate accessible name
+    const hasLabelContent = el.textContent !== '';
+    const inputLabelId = inputId + '-lbl';
+
+    // The host element must have a checkbox role to ensure proper VoiceOver
+    // support in Safari for accessibility
     return (
       <Host
+        role="checkbox"
         aria-checked={indeterminate ? 'mixed' : `${checked}`}
         aria-describedby={this.getHintTextID()}
         aria-invalid={this.getHintTextID() === this.errorTextId}
+        aria-labelledby={hasLabelContent ? inputLabelId : null}
+        aria-label={!hasLabelContent ? inheritedAttributes['aria-label'] || 'checkbox' : null}
```

Comment:
It would be best to not give it a default value like `checkbox` since apps can be in a different language and the dev might not pass an `aria-label`. What about?

```suggestion
        aria-label={inheritedAttributes['aria-label'] || null}
```

Reply:
You're required to provide a default label or it fails accessibility validation